Add x86_64 intrinsics guard

dmm-fb opened the pull request:
Test Plan: Run tests

Differential Revision: D38964321

albanD commented on c10/util/llvmMathExtras.h:
```diff
@@ -45,7 +45,9 @@
 #ifdef _MSC_VER
 // Declare these intrinsics manually rather including intrin.h. It's very
 // expensive, and MathExtras.h is popular.
+#if defined(__x86_64__) || defined(__i386__) || defined(_MSC_VER)
```
Not needed here haha

dmm-fb replied:
I'll fix it. Thanks

albanD reviewed (commented):
Looks good to me. 
cc @kimishpatel for the quantization sse code

dmm-fb committed:
Add x86_64 intrinsics guard (#83997)

Summary: Pull Request resolved: https://github.com/pytorch/pytorch/pull/83997

Test Plan: Run tests

Differential Revision: D38964321

fbshipit-source-id: 541c88939b9d7893aa7ad6f7e46c9381058c13ff

kimishpatel commented on aten/src/ATen/native/quantized/cpu/qnnpack/src/q8avgpool/up8x9-sse2.c:
```diff
@@ -8,7 +8,9 @@
 
 #include <assert.h>
 
+#if defined(__x86_64__) || defined(__i386__) || defined(_MSC_VER)
```
Is _MSC_VER necessary?

dmm-fb replied:
I think we can compile the code under Windows too - I don't think the __x86_64__ is defined there

kimishpatel replied:
My worry was compiling windows on arm. So something like `defined(_MSC_VER) && defined(_M_IX86)`? I am not sure if thats the right macro though.

kimishpatel reviewed (changes requested):
Looks good for the most part. Left one comment. How did you validate this?

cc: @digantdesai

ajtulloch commented:
@kimishpatel @digantdesai are your concerns addressed? Please accept if so.

ajtulloch reviewed (approved):
Looks reasonable.

malfet reviewed (approved):
Sure, why not, though, do we really need to check for `defined(__i386__)`?
And wouldn't it be better to define  `C10_X86` in `c10/macros/Macros.h` and use it throughout the code?

And by the way, a way of testing it would be to enable QNNPACK compilation for M1 target

kimishpatel reviewed (approved):
Left a comment but generally ok to land.
